tests/test_verifier.py

```
@@ -73,4 +74,29 @@ def test_verifier(self, m_resolver, m_deliver):
         addr = self.verifier._parse_address("user@example.com")
         m_resolver.assert_called_with('example.com', 'MX')
         m_deliver.assert_called_once_with(dns_response[0].to_text().split(), addr)
+
+    @patch.object(Verifier, '_can_deliver', side_effect=smtplib.SMTPHeloError(550, b'Blocked'))
+    @patch('dns.resolver.query', return_value=dns_response)
+    def test_verifier_returns_dict_on_smtp_helo_error(self, m_resolver, m_deliver):
```

Ruff (0.15.2)

[warning] Unused method argument: `m_resolver`

(ARG002)